--- tabbed/tabbed.c	Thu Feb 27 03:14:43 2020
+++ tabbed/tabbed.c	Thu Feb 27 03:15:27 2020
@@ -975,7 +975,7 @@
 void
 setup(void)
 {
-	int bitm, tx, ty, tw, th, dh, dw, isfixed;
+	int bitm, tx, ty, tw, th, dh, dw;
 	XWMHints *wmh;
 	XClassHint class_hint;
 	XSizeHints *size_hint;
@@ -1004,7 +1004,6 @@
 	wy = 0;
 	ww = 800;
 	wh = 600;
-	isfixed = 0;
 
 	if (geometry) {
 		tx = ty = tw = th = 0;
@@ -1022,8 +1021,6 @@
 			wx = -1;
 		if (bitm & YNegative && wy == 0)
 			wy = -1;
-		if (bitm & (HeightValue | WidthValue))
-			isfixed = 1;
 
 		dw = DisplayWidth(dpy, screen);
 		dh = DisplayHeight(dpy, screen);
--- tabbed/tabbed.c	2020-06-12 14:57:52.957622683 -0500
+++ tabbed/tabbed.c	2020-06-12 14:58:17.397855818 -0500
@@ -1063,16 +1063,11 @@
 	XSetClassHint(dpy, win, &class_hint);
 
 	size_hint = XAllocSizeHints();
-	if (!isfixed) {
-		size_hint->flags = PSize | PMinSize;
-		size_hint->height = wh;
-		size_hint->width = ww;
-		size_hint->min_height = bh + 1;
-	} else {
-		size_hint->flags = PMaxSize | PMinSize;
-		size_hint->min_width = size_hint->max_width = ww;
-		size_hint->min_height = size_hint->max_height = wh;
-	}
+
+	size_hint->flags = PMaxSize | PMinSize;
+	size_hint->min_width = size_hint->max_width = ww;
+	size_hint->min_height = size_hint->max_height = wh;
+
 	wmh = XAllocWMHints();
 	XSetWMProperties(dpy, win, NULL, NULL, NULL, 0, size_hint, wmh, NULL);
 	XFree(size_hint);
