From f0792e4daf566be0304aad18e43d16c044844f16 Mon Sep 17 00:00:00 2001
From: Georgios Oxinos <georgios.oxinos.extern@elinvar.de>
Date: Sun, 24 Jan 2021 16:38:28 +0100
Subject: [PATCH] [dwm][patch] patch that adds gaps to column layout

---
 config.def.h |  3 +++
 dwm.c        | 29 +++++++++++++++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/dwm.c b/dwm.c
index 4465af1..e72e00a 100644
--- dwm/dwm.c
+++ dwm/dwm.c
@@ -1670,6 +1670,35 @@ tagmon(const Arg *arg)
 	sendmon(selmon->sel, dirtomon(arg->i));
 }
 
+void
+col(Monitor *m) {
+  unsigned int i, n, h, w, x, y, mw;
+  Client *c;
+
+  for(n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
+  if(n == 0)
+    return;
+
+  if(n > m->nmaster)
+    mw = m->nmaster ? m->ww * m->mfact : 0;
+  else
+    mw = m->ww - m->gappx;
+
+  for(i = 0, x = y = m->gappx, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++) {
+    if(i < m->nmaster) {
+      w = (mw - x) / (MIN(n, m->nmaster) - i);
+      resize(c, x + m->wx, m->wy + m->gappx, w - (2*c->bw), m->wh - (2*c->bw) - 2*m->gappx, False);
+      if (x + WIDTH(c) + m->gappx < m->ww)
+        x += WIDTH(c) + m->gappx;
+    } else {
+      h = (m->wh - y) / (n - i) - m->gappx;
+      resize(c, x + m->wx, m->wy + y, m->ww - x - (2*c->bw) - m->gappx, h - (2*c->bw), False);
+      if (y + HEIGHT(c) + m->gappx < m->wh)
+        y += HEIGHT(c) + m->gappx;
+    }
+  }
+}
+
 void
 tile(Monitor *m)
 {
-- 
2.27.0

