--- dwm/dwm.c
+++ dwm/dwm.c
@@ -234,6 +234,7 @@
 static void spawn(const Arg *arg);
 static void tag(const Arg *arg);
 static void tile(Monitor *);
+static void col(Monitor *);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
 static void togglescratch(const Arg *arg);
@@ -1670,6 +1670,35 @@ tagmon(const Arg *arg)
 	sendmon(selmon->sel, dirtomon(arg->i));
 }
 
+void
+col(Monitor *m) {
+  unsigned int i, n, h, w, x, y, mw;
+  Client *c;
+
+  for(n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
+  if(n == 0)
+    return;
+
+  if(n > m->nmaster)
+    mw = m->nmaster ? m->ww * m->mfact : 0;
+  else
+    mw = m->ww - m->gappx;
+
+  for(i = 0, x = y = m->gappx, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++) {
+    if(i < m->nmaster) {
+      w = (mw - x) / (MIN(n, m->nmaster) - i);
+      resize(c, x + m->wx, m->wy + m->gappx, w - (2*c->bw), m->wh - (2*c->bw) - 2*m->gappx, False);
+      if (x + WIDTH(c) + m->gappx < m->ww)
+        x += WIDTH(c) + m->gappx;
+    } else {
+      h = (m->wh - y) / (n - i) - m->gappx;
+      resize(c, x + m->wx, m->wy + y, m->ww - x - (2*c->bw) - m->gappx, h - (2*c->bw), False);
+      if (y + HEIGHT(c) + m->gappx < m->wh)
+        y += HEIGHT(c) + m->gappx;
+    }
+  }
+}
+
 void
 tile(Monitor *m)
 {
