--- dwm/dwm.c
+++ dwm/dwm.c
@@ -204,7 +204,6 @@
 static void motionnotify(XEvent *e);
 static void movemouse(const Arg *arg);
 static Client *nexttiled(Client *c);
-static void pop(Client *);
 static void propertynotify(XEvent *e);
 static void quit(const Arg *arg);
 static Monitor *recttomon(int x, int y, int w, int h);
@@ -1274,15 +1273,6 @@
 {
 	for (; c && (c->isfloating || !ISVISIBLE(c)); c = c->next);
 	return c;
-}
-
-void
-pop(Client *c)
-{
-	detach(c);
-	attachabove(c);
-	focus(c);
-	arrange(c->mon);
 }
 
 void
