// http://github.com/mitchweaver/suckless
--- surf/surf.c	Thu Feb 27 18:34:09 2020
+++ surf/surf.c	Thu Feb 27 18:36:02 2020
@@ -241,7 +241,6 @@
 /* Buttons */
 static void clicknavigate(Client *c, const Arg *a, WebKitHitTestResult *h);
 static void clicknewwindow(Client *c, const Arg *a, WebKitHitTestResult *h);
-static void clickexternplayer(Client *c, const Arg *a, WebKitHitTestResult *h);
 
 static char winid[64];
 static char togglestats[12];
@@ -1982,15 +1981,6 @@
 
 	arg.v = webkit_hit_test_result_get_link_uri(h);
 	newwindow(c, &arg, a->i);
-}
-
-void
-clickexternplayer(Client *c, const Arg *a, WebKitHitTestResult *h)
-{
-	Arg arg;
-
-	arg = (Arg)VIDEOPLAY(webkit_hit_test_result_get_media_uri(h));
-	spawn(c, &arg);
 }
 
 int
