--- dwm.c.orig	2011-12-19 16:02:46.000000000 +0100
+++ dwm.c	2013-03-27 15:26:17.810041478 +0100
@@ -62,8 +62,8 @@
        NetWMFullscreen, NetActiveWindow, NetWMWindowType,
        NetWMWindowTypeDialog, NetLast };     /* EWMH atoms */
 enum { WMProtocols, WMDelete, WMState, WMTakeFocus, WMLast }; /* default atoms */
-enum { ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle,
-       ClkClientWin, ClkRootWin, ClkLast };             /* clicks */
+enum { ClkTagBar, ClkLtSymbol, ClkStatusText, ClkClientWin, ClkRootWin, ClkLast };             /* clicks */

 typedef union {
 	int i;
@@ -449,10 +449,8 @@
 		}
 		else if(ev->x < x + blw)
 			click = ClkLtSymbol;
-		else if(ev->x > selmon->ww - TEXTW(stext)) click = ClkStatusText;
 		else
-			click = ClkWinTitle;
+			click = ClkStatusText;
 	}
 	else if((c = wintoclient(ev->window))) {
 		focus(c);
@@ -753,13 +751,7 @@
 		dc.x = m->ww;
 	if((dc.w = dc.x - x) > bh) {
 		dc.x = x;
-		if(m->sel) {
-			col = m == selmon ? dc.sel : dc.norm;
-			drawtext(m->sel->name, col, False);
-			drawsquare(m->sel->isfixed, m->sel->isfloating, False, col);
-		}
-		else
-			drawtext(NULL, dc.norm, False);
+		drawtext(NULL, dc.norm, False);
 	}
 	XCopyArea(dpy, dc.drawable, m->barwin, dc.gc, 0, 0, m->ww, bh, 0, 0);
 	XSync(dpy, False);
@@ -1313,11 +1305,8 @@
 			drawbars();
 			break;
 		}
-		if(ev->atom == XA_WM_NAME || ev->atom == netatom[NetWMName]) {
+		if(ev->atom == XA_WM_NAME || ev->atom == netatom[NetWMName])
 			updatetitle(c);
-			if(c == c->mon->sel)
-				drawbar(c->mon);
-		}
 		if(ev->atom == netatom[NetWMWindowType])
 			updatewindowtype(c);
 	}
