--- sowm/Makefile   Fri Mar 20 10:26:50 2020
+++ sowm/Makefile	Fri Mar 20 10:27:35 2020
@@ -1,7 +1,8 @@
-CFLAGS += -std=c99 -Wall -Wextra -pedantic -Wold-style-declaration
-CFLAGS += -Wmissing-prototypes -Wno-unused-parameter
+CFLAGS += -std=c99 -Wall -Wextra -pedantic
 PREFIX ?= /usr
 BINDIR ?= $(PREFIX)/bin
+LIBS   += -L/usr/X11R6/lib -lX11
+INCS   += -I/usr/X11R6/include
 CC     ?= gcc
 
 all: config.h sowm
@@ -10,7 +11,7 @@
 	cp config.def.h config.h
 
 sowm:
-	$(CC) -O3 $(CFLAGS) -o sowm sowm.c -lX11 $(LDFLAGS)
+	$(CC) -O3 $(CFLAGS) -o sowm sowm.c $(LIBS) $(INCS) $(LDFLAGS)
 
 install: all
 	install -Dm755 sowm $(DESTDIR)$(BINDIR)/sowm
