// http://github.com/mitchweaver/suckless
--- surf/surf.c	Thu Feb 27 19:06:17 2020
+++ surf/surf.c	Thu Feb 27 19:06:55 2020
@@ -231,7 +231,6 @@
 static void navigate(Client *c, const Arg *a);
 static void stop(Client *c, const Arg *a);
 static void togglefullscreen(Client *c, const Arg *a);
-static void togglecookiepolicy(Client *c, const Arg *a);
 static void toggleinspector(Client *c, const Arg *a);
 static void find(Client *c, const Arg *a);
 
@@ -1881,15 +1880,6 @@
 		gtk_window_unfullscreen(GTK_WINDOW(c->win));
 	else
 		gtk_window_fullscreen(GTK_WINDOW(c->win));
-}
-
-void
-togglecookiepolicy(Client *c, const Arg *a)
-{
-	++cookiepolicy;
-	cookiepolicy %= strlen(curconfig[CookiePolicies].val.v);
-
-	setparameter(c, 0, CookiePolicies, NULL);
 }
 
 void
