diff --git a/surf.c b/surf.c
index 93a1629..c20537e 100644
--- a/surf.c
+++ b/surf.c
@@ -476,6 +476,8 @@ loaduri(Client *c, const Arg *a)
 	} else if (!stat(uri, &st) && (path = realpath(uri, NULL))) {
 		url = g_strdup_printf("file://%s", path);
 		free(path);
+	} else if (*uri == ' ') {
+		url = g_strdup_printf("%s%s", searchengine, uri + 1);
 	} else {
 		url = g_strdup_printf("http://%s", uri);
 	}
